### DISCUSSION TITLE
Rewrite links from imported MDX/Markdown files

### DISCUSSION THREAD

**Discussion by tizmagik** [facebook/docusaurus#5885] [Q&A]
I searched but couldn't find anything about this, so apologies if this is already covered somewhere...

I'm using the [Markdown import feature described here](https://docusaurus.io/docs/markdown-features/react#importing-markdown),

```
---
title: Other
hide_title: true
---

import OtherReadme from '../../../../apps/Other/README.md';

<OtherReadme />
```

This is so that we can co-locate those READMEs with the source-code it documents and acts as a landing page while navigating the codebase on Github, but at the same time include it in our published Docusaurus docs site.

However, because that README actually lives _outside_ of the Docusaurus docs folder, any links to source-code or other READMEs are broken. Presumably, it should be possible to rewrite those links, but not sure how best to do that? Do I need to write a custom plugin? Would it be a Remark plugin? Or is there some other Docusaurus configuration that would be useful here?

Here's an example of a broken link reported:

```
warn Docusaurus found broken links!

Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.

Exhaustive list of all broken links found:

- On source page path = /docs/apps/Other:
   -> linking to ../../packages/@ecomm/toggles/README.md (resolved as: /packages/@ecomm/toggles/README.md)
```

Any tips on how to rewrite those links would be appreciated, thank you!

**Comment by slorber** [comment]
I think a Remark plugin could do the job yes, not sure anyone already did that in practice so you'll have to figure it out 😅

**Reply by tizmagik** [reply]
Thanks for your response @slorber ! So I've tried writing a remark and a rehype plugin that just outputs all links (remark: node of type `link`; rehype: element with tagName `a`)  and configuring Docusaurus with it but the only links I see are the ones that are statically available. It looks like these plugins do not have visibility into the links that are _inside_ of Markdown that is imported as described in the original post.

I also tried specifying a component override on import [as described in MDX docs](https://mdxjs.com/table-of-components/#components):

```markdown
import OtherReadme from '../../../../apps/Other/README.md';
import LinkFixer from './LinkFixer.js';

<OtherReadme components={{a: LinkFixer }} />
```

But that didn't work. It doesn't look like my `LinkFixer` component ends up getting used.


It's almost as if I'd need to pass in some additional plugins or something when importing, something like:

```markdown
import OtherReadme from '../../../../apps/Other/README.md';

<OtherReadme plugins={[myUrlRewritePlugin]} />
```

Any ideas?

**Reply by slorber** [reply]
Have you tried `beforeDefaultRemarkPlugins` option?

https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-pages#beforeDefaultRemarkPlugins

We have a `@theme/MDXComponents` that you can try to swizzle too if you want to create your own custom md link

**Comment by luckyluggi** [comment]
Is there already a solution for this problem?
